Sources/Combine/Extensions/UIAlertController+Combine.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  UIAlertController+Combine.swift
+//  Catalog
+//
+//  Created by Zvonimir Medak on 24.11.2021..
+//  Copyright © 2021 Infinum. All rights reserved.
+//
+
+import UIKit
+import Combine
+
+/// Combine alert action wrapper
+@available(iOS 13, *)
+public struct CombineAlertAction<T> {
+
+    /// Action title
+    public let title: String
+    /// Action style
+    public let style: UIAlertAction.Style
+    /// Value which will be returned upon confirming action
+    public let value: T?
+
+    public init(title: String, style: UIAlertAction.Style = .default, value: T? = nil) {
+        self.title = title
+        self.style = style
+        self.value = value
+    }
+
+    func asUIAlertAction(with completed: @escaping (Result<T?, Never>) -> Void) -> UIAlertAction {
+        return UIAlertAction(title: title, style: style) { _ in
+            guard let value = self.value else {
+                completed(.success(nil))
+                return
+            }
+            completed(.success(value))
+        }
```

Comment:
Since you're returning success in both cases and `completed` is accepting an optional, just having this is enough 🙂 

Also, unown/weakify self 🙂 

```suggestion
        return UIAlertAction(title: title, style: style) { [unowned self] _ in
            completed(.success(value))
        }
```